publication-commons/src/main/java/no/unit/nva/publication/external/services/PersonApiClient.java

```diff
@@ -0,0 +1,109 @@
+import org.zalando.problem.Status;
+
+public class PersonApiClient {
+
+    public static final HttpClient HTTP_CLIENT = HttpClient.newBuilder().build();
+    public static final String ERROR_WITH_PERSON_SERVICE_COMMINUCATION = "Error communicating with Person service";
+    public static final String PERSON_SERVICE_RESPONSE = "Person Service response:";
+    public static final String USER_AFFILIATIONS_FIELD = "/orgunitids";
+    public static final String PATH_TO_PESON_SERVICE_PROXY = "person";
+    private static final Logger logger = LoggerFactory.getLogger(PersonApiClient.class);
+    private final HttpClient httpClient;
+
+    public PersonApiClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    @JacocoGenerated
+    public PersonApiClient() {
+        this.httpClient = HTTP_CLIENT;
+    }
+
+    public List<URI> fetchAffiliationsForUser(String feideId)
+        throws IOException, InterruptedException, ApiGatewayException {
+        var queryUri = createQueryToPersonService(feideId);
+        var response = sendQuery(queryUri);
+        checkResponse(response);
+        return extractUserAffiliations(response);
+    }
+
+    private void checkResponse(HttpResponse<String> response) throws BadGatewayException {
+        if (!Objects.equals(HttpURLConnection.HTTP_OK, response.statusCode())) {
+            var errorReport = generateErrorReport(response);
+            logger.warn(PERSON_SERVICE_RESPONSE + errorReport);
+            throw new BadGatewayException(ERROR_WITH_PERSON_SERVICE_COMMINUCATION);
+        }
+    }
+
+    private Problem generateErrorReport(HttpResponse<String> response) {
+        return Problem.builder()
+            .withStatus(Status.valueOf(response.statusCode()))
+            .withDetail(response.body())
+            .build();
+    }
+
+    private HttpResponse<String> sendQuery(URI queryUri) throws IOException, InterruptedException {
+        var httpRequest = HttpRequest.newBuilder(queryUri)
+            .header(ACCEPT, MediaType.JSON_UTF_8.toString())
+            .GET()
+            .build();
+        return httpClient.send(httpRequest, BodyHandlers.ofString(StandardCharsets.UTF_8));
+    }
+
+    private URI createQueryToPersonService(String feideId) {
+        return new UriWrapper("https", API_HOST)
+            .addChild(PATH_TO_PESON_SERVICE_PROXY)
+            .addQueryParameter("feideid", feideId)
+            .getUri();
+    }
+
+    private List<URI> extractUserAffiliations(HttpResponse<String> response) throws JsonProcessingException {
+        return extractUserInformation(response)
+            .map(details -> (ArrayNode) details.at(USER_AFFILIATIONS_FIELD))
+            .map(this::toUriList)
+            .orElse(Collections.emptyList());
+    }
+
+    private Optional<ObjectNode> extractUserInformation(HttpResponse<String> response) throws JsonProcessingException {
+        var result = (ArrayNode) dtoObjectMapper.readTree(response.body());
```

Comment:
The response is always a list, right? IIRC, there is a way around this, but perhaps not relevant here